plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt

```diff
@@ -194,7 +194,7 @@ class CocoaPods(
 
         val podspecCommand = runCatching {
             run(
-                "spec", "which", podspecName,
+                "spec", "which", "^$podspecName$",
                 "--version=${id.version}",
                 "--allow-root",
                 "--regex",
```

Comment:
Just wondering, could we simply drop `--regex`, as I assume `podspecName` is always a literal string in our case?

Reply:
> Just wondering, could we simply drop `--regex`, as I assume `podspecName` is always a literal string in our case?

I also experimented with removing the --regex option, but the result remained the same when I ran the following command:
```
$ pod spec which "coaSec" --version=1.2.4 --allow-root
/Users/REDACTED/.cocoapods/repos/cocoapods/Specs/d/e/9/CocoaSecurity/1.2.4/CocoaSecurity.podspec.json
```
I attempted to explore alternative options by using `pod spec which --help`, but I couldn't find any other options that seemed relevant to our situation.
```
$ pod spec which --help
Usage:

    $ pod spec which [QUERY]

      Prints the path of the .podspec file(s) whose name matches `QUERY`

Options:

    --regex        Interpret the `QUERY` as a regular expression
    --show-all     Print all versions of the given podspec
    --version      Print a specific version of the given podspec
    --allow-root   Allows CocoaPods to run as root
    --silent       Show nothing
    --verbose      Show more debugging information
    --no-ansi      Show output without ANSI codes
    --help         Show help banner of specified command
```

Reply:
> but the result remained the same when I ran the following command:

But that's good, isn't it? If the result does not change without `--regex`, it means we can drop it, and avoid the need to make the regex more strict, right?

Reply:
> > but the result remained the same when I ran the following command:
> 
> But that's good, isn't it? If the result does not change without `--regex`, it means we can drop it, and avoid the need to make the regex more strict, right?

What I meant by saying that the result remained the same is that it consistently produced an incorrect result. 
```
$ pod spec which "coaSec" --version=1.2.4 --allow-root
/Users/REDACTED/.cocoapods/repos/cocoapods/Specs/d/e/9/CocoaSecurity/1.2.4/CocoaSecurity.podspec.json
-> Resulted in a mismatch where you expected "coaSec," but it returned "CocoaSecurity."
```
To clarify, when I tested it, I found that in order to accurately match the exact string I was looking for, I needed to use `regex` checks with ^ and $.

Reply:
But when run from ORT, `pod spec which` gets exclusively called with an `Identifier` name, which can never be a regex. Thus IMO the `--regex` is unneeded. And if it's not needed, also you stricter regex matching is not needed.

Reply:
Here's my proposal for that: https://github.com/oss-review-toolkit/ort/pull/7552

Reply:
Ah, now I believe to understand what you're saying. So the rationale for a proper commit message would be:

```
fix(CocoaPods): Restrict package name match to full matches

The both the default and the regex query string matching logic actually
match substrings case-insensitively. So if a package name is a substring of
another package name, invalid matches could occur. To circumvent that,
force the regex to be a full match.
```

If you agree with the rational, please copy & paste that to your commit as the message.

Reply:
> Ah, now I believe to understand what you're saying. So the rationale for a proper commit message would be:
> 
> ```
> fix(CocoaPods): Restrict package name match to full matches
> 
> The both the default and the regex query string matching logic actually
> match substrings case-insensitively. So if a package name is a substring of
> another package name, invalid matches could occur. To circumvent that,
> force the regex to be a full match.
> ```
> 
> If you agree with the rational, please copy & paste that to your commit as the message.

I edited my commit message it as you suggested!

Reply:
I'm not sure about how controversial the other commit is, but this one now is IMO good to go.
We could split-out this commit to a separate PR and merge it already now? (makes sense @sschuberth ?)